tests/unit/backend/survey.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { expect, jest } from "@jest/globals"
+import { SurveyType } from "../../../lib/enums/survey"
+import { Survey } from "@lib/types/survey"
+import FollowupModel from "../../../backend/models/followup.js"
+
+describe("Survey methods", () => {
+  let mockFollowup
+
+  beforeEach(() => {
+    mockFollowup = {
+      surveys: [],
+      save: jest.fn(),
+      addSurveyIfMissing: jest.fn(),
+      updateSurvey: FollowupModel.prototype.updateSurvey,
+    }
+  })
+
+  describe("addSurveyIfMissing", () => {
+    it("should add survey if not exists", async () => {
+      const existingSurvey: Survey = {
+        _id: "test-id",
+        type: SurveyType.TrackClickOnSimulationUsefulnessEmail,
+        answers: [],
+        touchedAts: [],
+      }
+      mockFollowup.surveys = [existingSurvey]
+      const newAnswers = [{ id: "wasUseful", value: true }]
+      await mockFollowup.updateSurvey(
+        SurveyType.TrackClickOnSimulationUsefulnessEmail,
+        newAnswers
+      )
+      expect(mockFollowup.surveys[0].answers).toEqual(newAnswers)
+      expect(mockFollowup.surveys[0].touchedAts).toHaveLength(1)
+    })
```

Comment:
Cette méthode n'ajoute pas de `touchedAts`. On poursuit avec l'appel de la méthode `addSurveyIfMissing` du model

```suggestion
    it("should add a new survey if it doesn't exist", async () => {
      const result = await mockFollowup.addSurveyIfMissing(
        SurveyType.TrackClickOnSimulationUsefulnessEmail
      )

      expect(result.type).toEqual(
        SurveyType.TrackClickOnSimulationUsefulnessEmail
      )
      expect(mockFollowup.surveys).toHaveLength(1)
      expect(mockFollowup.surveys[0].type).toEqual(
        SurveyType.TrackClickOnSimulationUsefulnessEmail
      )
    })
```

Reply:
Avec ta proposition, on supprime le test de update. Mais sur le fond tu as raison, je vais réadapter l'intégration de la fonction.